dj/api/nodes.py

```diff
@@ -176,46 +87,108 @@ def validate_node(
     )
 
 
-@router.get("/nodes/", response_model=List[NodeMetadata])
-def read_nodes(*, session: Session = Depends(get_session)) -> List[NodeMetadata]:
+@router.get("/nodes/", response_model=List[NodeOutput])
+def read_nodes(*, session: Session = Depends(get_session)) -> List[NodeOutput]:
     """
     List the available nodes.
     """
     nodes = session.exec(select(Node).options(joinedload(Node.current))).unique().all()
     return nodes
 
 
-@router.get("/nodes/{name}/", response_model=NodeMetadata)
-def read_node(name: str, *, session: Session = Depends(get_session)) -> NodeMetadata:
+@router.get("/nodes/{name}/", response_model=NodeOutput)
+def read_node(name: str, *, session: Session = Depends(get_session)) -> NodeOutput:
     """
     Show the active version of the specified node.
     """
-    statement = select(Node).where(Node.name == name).options(joinedload(Node.current))
-    node = session.exec(statement).unique().one_or_none()
-    if not node:
-        raise HTTPException(
-            status_code=HTTPStatus.NOT_FOUND,
-            detail=f"Node not found: `{name}`",
-        )
+    node = get_node_by_name(session, name, with_current=True)
     return node  # type: ignore
 
 
-@router.get("/nodes/{name}/revisions/", response_model=List[NodeRevisionMetadata])
+@router.post("/nodes/{name}/materialization/")
+def upsert_node_materialization_config(
+    name: str,
+    data: UpsertMaterializationConfig,
+    *,
+    session: Session = Depends(get_session),
+) -> JSONResponse:
+    """
+    Update materialization config of the specified node.
+    """
+    node = get_node_by_name(session, name, with_current=True)
+    if node.type == NodeType.SOURCE:
+        raise DJException(
+            http_status_code=HTTPStatus.BAD_REQUEST,
+            message=f"Cannot set materialization config for source node `{name}`!",
+        )
+    current_revision = node.current
+
+    # Check to see if a config for this engine already exists with the exact same config
+    existing_config_for_engine = [
+        config
+        for config in node.current.materialization_configs
+        if config.engine.name == data.engine_name
+    ]
+    if (
+        existing_config_for_engine
+        and existing_config_for_engine[0].config == data.config
+    ):
+        return JSONResponse(
+            status_code=200,
```

Comment:
nit (and not sure how useful): Should we start to use 204 "No content" status codes for successful idempotent operations that didn't make any changes server side? I've seen this leveraged pretty nicely by front ends that see 204 and know to show a "toast" notification instead of going to a different page.

Reply:
Yeah makes sense. I went ahead and updated it to a 204.